From: Nick Rosbrook <nick.rosbrook@canonical.com>
Date: Tue, 21 Jun 2022 10:52:39 -0400
Subject: test: increase QEMU_MEM for some tests

These tests have a tendency to fail with OOM on the autopkgtest
infrastructure. Increase QEMU_MEM to try and alleviate that.
---
 test/TEST-13-NSPAWN/test.sh             | 2 ++
 test/TEST-17-UDEV/test.sh               | 2 ++
 test/TEST-19-CGROUP/test.sh             | 1 +
 test/TEST-24-CRYPTSETUP/test.sh         | 2 ++
 test/TEST-29-PORTABLE/test.sh           | 2 ++
 test/TEST-30-ONCLOCKCHANGE/test.sh      | 2 ++
 test/TEST-31-DEVICE-ENUMERATION/test.sh | 1 +
 test/TEST-32-OOMPOLICY/test.sh          | 1 +
 test/TEST-36-NUMAPOLICY/test.sh         | 2 ++
 test/TEST-38-FREEZER/test.sh            | 2 ++
 test/TEST-50-DISSECT/test.sh            | 2 ++
 test/TEST-53-ISSUE-16347/test.sh        | 2 ++
 test/TEST-58-REPART/test.sh             | 2 ++
 test/TEST-62-RESTRICT-IFACES/test.sh    | 2 ++
 test/TEST-66-DEVICE-ISOLATION/test.sh   | 2 ++
 test/TEST-67-INTEGRITY/test.sh          | 2 ++
 test/TEST-70-TPM2/test.sh               | 2 ++
 17 files changed, 31 insertions(+)

diff --git a/test/TEST-13-NSPAWN/test.sh b/test/TEST-13-NSPAWN/test.sh
index 2e94156..c6dd847 100755
--- a/test/TEST-13-NSPAWN/test.sh
+++ b/test/TEST-13-NSPAWN/test.sh
@@ -42,4 +42,6 @@ EOF
     chmod +x "$container/sbin/init"
 }
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-17-UDEV/test.sh b/test/TEST-17-UDEV/test.sh
index f7a9075..46c317d 100755
--- a/test/TEST-17-UDEV/test.sh
+++ b/test/TEST-17-UDEV/test.sh
@@ -13,4 +13,6 @@ test_append_files() {
     generate_module_dependencies
 }
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-19-CGROUP/test.sh b/test/TEST-19-CGROUP/test.sh
index ba05b5e..6418b71 100755
--- a/test/TEST-19-CGROUP/test.sh
+++ b/test/TEST-19-CGROUP/test.sh
@@ -7,6 +7,7 @@ TEST_DESCRIPTION="Various cgroup-related tests"
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
+QEMU_MEM="1024M"
 UNIFIED_CGROUP_HIERARCHY=yes
 
 do_test "$@"
diff --git a/test/TEST-24-CRYPTSETUP/test.sh b/test/TEST-24-CRYPTSETUP/test.sh
index d0ec63d..9810216 100755
--- a/test/TEST-24-CRYPTSETUP/test.sh
+++ b/test/TEST-24-CRYPTSETUP/test.sh
@@ -116,4 +116,6 @@ test_setup_cleanup() {
     cleanup_initdir
 }
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-29-PORTABLE/test.sh b/test/TEST-29-PORTABLE/test.sh
index d826db6..ab8876a 100755
--- a/test/TEST-29-PORTABLE/test.sh
+++ b/test/TEST-29-PORTABLE/test.sh
@@ -22,4 +22,6 @@ test_append_files() {
     install_verity_minimal
 }
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-30-ONCLOCKCHANGE/test.sh b/test/TEST-30-ONCLOCKCHANGE/test.sh
index 69dbdb2..66fdb39 100755
--- a/test/TEST-30-ONCLOCKCHANGE/test.sh
+++ b/test/TEST-30-ONCLOCKCHANGE/test.sh
@@ -8,4 +8,6 @@ TEST_NO_NSPAWN=1
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-31-DEVICE-ENUMERATION/test.sh b/test/TEST-31-DEVICE-ENUMERATION/test.sh
index 107fa7d..cbc46f3 100755
--- a/test/TEST-31-DEVICE-ENUMERATION/test.sh
+++ b/test/TEST-31-DEVICE-ENUMERATION/test.sh
@@ -8,5 +8,6 @@ TEST_NO_NSPAWN=1
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
+QEMU_MEM="1024M"
 
 do_test "$@"
diff --git a/test/TEST-32-OOMPOLICY/test.sh b/test/TEST-32-OOMPOLICY/test.sh
index 78c4e648..0cad078 100755
--- a/test/TEST-32-OOMPOLICY/test.sh
+++ b/test/TEST-32-OOMPOLICY/test.sh
@@ -9,5 +9,6 @@ TEST_NO_NSPAWN=1
 . "${TEST_BASE_DIR:?}/test-functions"
 
 UNIFIED_CGROUP_HIERARCHY=yes
+QEMU_MEM="1024M"
 
 do_test "$@"
diff --git a/test/TEST-36-NUMAPOLICY/test.sh b/test/TEST-36-NUMAPOLICY/test.sh
index 9548ebb..e873139 100755
--- a/test/TEST-36-NUMAPOLICY/test.sh
+++ b/test/TEST-36-NUMAPOLICY/test.sh
@@ -8,6 +8,8 @@ TEST_NO_NSPAWN=1
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
+QEMU_MEM="1024M"
+
 if qemu_min_version "5.2.0"; then
     QEMU_OPTIONS+=" -object memory-backend-ram,id=mem0,size=${QEMU_MEM:-768M} -numa node,memdev=mem0,nodeid=0"
 else
diff --git a/test/TEST-38-FREEZER/test.sh b/test/TEST-38-FREEZER/test.sh
index da7bfc6..8d4d0f0 100755
--- a/test/TEST-38-FREEZER/test.sh
+++ b/test/TEST-38-FREEZER/test.sh
@@ -8,4 +8,6 @@ TEST_NO_NSPAWN=1
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-50-DISSECT/test.sh b/test/TEST-50-DISSECT/test.sh
index f1abce8..42ed4e5 100755
--- a/test/TEST-50-DISSECT/test.sh
+++ b/test/TEST-50-DISSECT/test.sh
@@ -29,4 +29,6 @@ test_append_files() {
     install_verity_minimal
 }
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-53-ISSUE-16347/test.sh b/test/TEST-53-ISSUE-16347/test.sh
index 6d4821d..f68dc22 100755
--- a/test/TEST-53-ISSUE-16347/test.sh
+++ b/test/TEST-53-ISSUE-16347/test.sh
@@ -10,4 +10,6 @@ QEMU_OPTIONS+=" -rtc base=$(date -u +%Y-%m-%dT%H:%M:%S -d '+3 days')"
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-58-REPART/test.sh b/test/TEST-58-REPART/test.sh
index 14898b5..8b53d21 100755
--- a/test/TEST-58-REPART/test.sh
+++ b/test/TEST-58-REPART/test.sh
@@ -28,4 +28,6 @@ test_append_files() {
     fi
 }
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-62-RESTRICT-IFACES/test.sh b/test/TEST-62-RESTRICT-IFACES/test.sh
index db8290b..8e65388 100755
--- a/test/TEST-62-RESTRICT-IFACES/test.sh
+++ b/test/TEST-62-RESTRICT-IFACES/test.sh
@@ -8,4 +8,6 @@ TEST_NO_NSPAWN=1
 # shellcheck source=test/test-functions
 . "$TEST_BASE_DIR/test-functions"
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-66-DEVICE-ISOLATION/test.sh b/test/TEST-66-DEVICE-ISOLATION/test.sh
index 28eed2c..78dfed8 100755
--- a/test/TEST-66-DEVICE-ISOLATION/test.sh
+++ b/test/TEST-66-DEVICE-ISOLATION/test.sh
@@ -8,4 +8,6 @@ TEST_NO_NSPAWN=1
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
+QEMU_MEM="1024M"
+
 do_test "$@"
diff --git a/test/TEST-67-INTEGRITY/test.sh b/test/TEST-67-INTEGRITY/test.sh
index 1ec2d4b..944e15d 100755
--- a/test/TEST-67-INTEGRITY/test.sh
+++ b/test/TEST-67-INTEGRITY/test.sh
@@ -9,6 +9,8 @@ TEST_NO_NSPAWN=1
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
+QEMU_MEM="1024M"
+
 test_append_files() {(
 
     instmods loop =block
diff --git a/test/TEST-70-TPM2/test.sh b/test/TEST-70-TPM2/test.sh
index c33e71c..ef807f2 100755
--- a/test/TEST-70-TPM2/test.sh
+++ b/test/TEST-70-TPM2/test.sh
@@ -34,4 +34,6 @@ test_append_files() {
     inst_binary tpm2_readpublic
 }
 
+QEMU_MEM="1024M"
+
 do_test "$@"
